[release-1.30] Update operator version to 1.30.1

dsimansk opened the pull request:
Fixes JIRA #

Bump S-O version to next patch release. 

/cc @Kaustubh-pande @mgencur 

## Proposed Changes

<!-- Please categorize your changes:
- :gift: Add new feature
- :bug: Fix bug
- :broom: Update or clean up current behavior
- :wastebasket: Remove feature or internal logic
-->

- [release-1.30] Update operator version to 1.30.1

dsimansk commented on knative-operator/deploy/resources/knativekafka/controller/eventing-kafka-controller.yaml:
```diff
@@ -704,15 +704,22 @@ metadata:
     app.kubernetes.io/version: v1.9
     knative.dev/crd-install: "true"
-  name: consumers.internal.kafka.eventing.knative.dev
+  name: consumergroups.internal.kafka.eventing.knative.dev
```
@pierDipi @matzew is it safe to land this on 1.30 branch? This is produced by `make generated-files` and flagged by validation check as missing. But I'd rather double check.

dsimansk replied:
OK it seems like an ordering swap to me. I hope that's fine then. :)

pierDipi replied:
Yes

dsimansk replied:
Thanks for confirming that.

Kaustubh-pande commented:
/retest

mgencur commented on olm-catalog/serverless-operator/index/configs/index.yaml:
```diff
@@ -3,34 +3,34 @@ schema: olm.channel
   - name: "serverless-operator.v1.30.0"
     replaces: "serverless-operator.v1.29.0"
     skipRange: ">=1.29.0 <1.30.0"
+  - name: "serverless-operator.v1.30.1"
```
Just a note for the future because we have been setting it wrong for this stable-1.29 channel. It should include versions up to 1.29.x. Not 1.30 or newer.

dsimansk replied:
I've created a new issue from your comment to have captured outside of the PR. But I assume it's doesn't break anything yet.

mgencur commented on olm-catalog/serverless-operator/project.yaml:
```diff
@@ -73,3 +73,5 @@ upgrade_sequence:
       source: redhat-operators
     - csv: serverless-operator.v1.30.0
       source: serverless-operator
```
```suggestion
      source: redhat-operators
```

mgencur replied:
1.30 is now live so we want to install it from redhat-operators

mgencur commented on olm-catalog/serverless-operator/project.yaml:
```diff
@@ -4,19 +4,19 @@ project:
+        skipRange: '>=1.29.0 <1.30.0'
 requirements:
     kube:
         # The min version validation in `vendor/knative.dev/pkg/version/version.go`
```
We're not bumping serving and eventing so stuff under `dependencies.previous` should have same versions as the latest:
```
dependencies:
   previous:
        serving: knative-v1.9
        eventing: knative-v1.9
        eventing_kafka_broker: knative-v1.9
```

mgencur replied:
Downgrading Serverless should then pass.

mgencur reviewed (changes requested):
Hey @dsimansk , a couple of comments. After addressing them we should see upgrade tests passing, IMO.
I've seen you already bumped `promotion` in openshift/release repo. Great.

dsimansk commented:
> Hey @dsimansk , a couple of comments. After addressing them we should see upgrade tests passing, IMO. I've seen you already bumped `promotion` in openshift/release repo. Great.

Thanks for the version pointers, Martin! I've addressed your comments now, so hopefully upgrade tests are in better shape now. 🤞

dsimansk commented:
/test 4.12-operator-e2e-aws-ocp-412

dsimansk commented:
/test 4.12-upgrade-tests-aws-ocp-412

dsimansk commented:
/test 4.12-upgrade-tests-aws-ocp-412

dsimansk commented:
/test 4.12-upstream-e2e-mesh-aws-ocp-412

dsimansk commented:
/test 4.12-upstream-e2e-mesh-aws-ocp-412

dsimansk commented:
The upgrade error seems to be related to the zipkin setup issues fixed after 1.30 branch cut on the `main`. https://github.com/openshift-knative/serverless-operator/pull/2255

@mgencur could you pls check the PR again, if it's ok to go. Thanks!

dsimansk commented:
/test 4.12-upgrade-tests-aws-ocp-412
/test 4.12-upstream-e2e-mesh-aws-ocp-412

Kaustubh-pande commented:
/lgtm
As required tests are already passed.
